ckanext/switzerland/plugins.py

```diff
@@ -429,10 +429,14 @@ def before_update(self, mapper, connection, package_extra):
         This prevents future datasets with the same name will not have a
         number appended.
         """
-        if "deleted" == package_extra.state:
-            package_extra.name = self._ensure_name_is_unique(
+        try:
+            if "deleted" == package_extra.state\
+                    and hasattr(package_extra, 'name'):
```

Comment:
Just curious: Does it often happen that `extras` has the given named attribute 'name'? 
I have checked few datasets at opendata.swiss and did not find any "name" in `extras`

Reply:
in this context, `package_extra` is not `package_extras`. When using this interface we get the package in a dict which is called `package_extra` so we expect it to have a name in most cases :)

Reply:
I see, thank you :)